frontend/src/FullPageRoutes/Profile.tsx

```diff
@@ -19,58 +32,82 @@ const useStyles = makeStyles({
   },
 })
 
-type UserProfile = {
-  first_name: string
-  last_name: string
-  phone_number?: string
-  age?: number
-  gender: string
-}
-
 export default function Profile(): JSX.Element {
+  const [isEditMode, toggleEditMode] = React.useState<boolean>(false)
+  const [errMessage, seterrMessage] = React.useState<string>("")
   const classes = useStyles()
 
-  const [username, setUsername] = React.useState<string>("")
-  React.useEffect(() => {
-    request<UserProfile>({ path: "/api/profile/", method: "get" }).then(
-      (profile) => {
-        setUsername(profile.parsedBody.first_name)
-      }
-    )
+  const [readonlyContent, setProfile] = React.useState<ProfileContentType>({
+    username: "",
+    firstName: "",
+    lastName: "",
+    birthday: new Date(),
+    gender: "",
+    interests: "",
   })
+  const { editableContent, setters } = useProfile(readonlyContent)
 
-  // Dummy data.
-  const content = {
-    username: username,
-    name: "Tom Cruise",
-    profileImg: "",
-    gender: "Male",
-    religion: "Scientology",
-    location: "Hollywood",
-    occupations: ["All the above"],
-    age: "58",
-    interests: ["Acting", "Film Producing"],
+  React.useEffect(() => {
+    request<UserProfile>({ path: "/api/profile/", method: "get" })
+      .then((res) => {
+        const profile: UserProfile = res.parsedBody
+        const content: ProfileContentType = {
+          username: profile.username,
+          firstName: profile.first_name,
+          lastName: profile.last_name,
+          birthday: new Date(profile.birth_date),
+          gender: profile.gender,
+          interests: profile.interests,
+        }
+        setProfile(content)
+      })
+      .catch((err) => {
+        console.log(err)
+      })
+  }, [])
```

Comment:
should we add a dependency here like readonlyContent?

Reply:
I keep running into a disparity between what React documentation says and what I see in practice. The bug where render is called constantly is supposedly fixed by adding dependencies or the empty `[]`. However, I found that the bug was present with the two dependencies and not present with the empty `[]`.